Content.Server/Administration/Managers/WatchlistWebhookManager.cs

```diff
@@ -0,0 +1,166 @@
+using Content.Server.Administration.Notes;
+using Content.Server.Database;
+using Content.Server.Discord;
+using Content.Shared.CCVar;
+using Robust.Server;
+using Robust.Server.Player;
+using Robust.Shared.Enums;
+using Robust.Shared.Configuration;
+using Robust.Shared.Network;
+using Robust.Shared.Player;
+using Robust.Shared.Timing;
+using System.Linq;
+
+namespace Content.Server.Administration.Managers;
+
+/// <summary>
+///     This manager sends a Discord webhook notification whenever a player with an active
+///     watchlist joins the server.
+/// </summary>
+public sealed class WatchlistWebhookManager : IWatchlistWebhookManager
+{
+    [Dependency] private readonly IAdminNotesManager _adminNotes = default!;
+    [Dependency] private readonly IBaseServer _baseServer = default!;
+    [Dependency] private readonly IConfigurationManager _cfg = default!;
+    [Dependency] private readonly DiscordWebhook _discord = default!;
+    [Dependency] private readonly IPlayerManager _playerManager = default!;
+
+    private ISawmill _sawmill = default!;
+
+    private WebhookIdentifier? _webhookIdentifier;
+    private List<WatchlistConnection> watchlistConnections = new();
+    private TimeSpan _bufferTime;
+    private RStopwatch? _bufferingStopwatch;
+
+    public void Initialize()
+    {
+        _sawmill = Logger.GetSawmill("discord");
+
+        var bufferTimeSeconds = _cfg.GetCVar(CCVars.DiscordWatchlistConnectionBufferTime);
+        _bufferTime = TimeSpan.FromSeconds(bufferTimeSeconds);
+
+        var webhook = _cfg.GetCVar(CCVars.DiscordWatchlistConnectionWebhook);
+        if (!string.IsNullOrWhiteSpace(webhook))
+        {
+            _discord.GetWebhook(webhook, data => _webhookIdentifier = data.ToIdentifier());
+            _playerManager.PlayerStatusChanged += OnPlayerStatusChanged;
+        }
```

Comment:
Didn't you have this using `Subs.CVar()` before? Any reason you stopped doing that?

Reply:
I couldn't figure out the right way to get `Subs` outside of an `EntitySystem` - the only place I could find in the codebase where `new Submissions()` is called is inside the base `EntitySystem` class, and I wasn't sure how to use it elsewhere, so I opted for less extra complexity. I presume the difference is that `Subs` will re-evaluate the passed closure if the CVars are changed after startup? It probably doesn't matter too much here since the webhook CVars are `SERVER_ONLY`.

Reply:
You need to use `IConfigurationManager.OnValueChanged`.